src/coefficients/stdfunction_coefficient.cpp

```diff
@@ -160,15 +160,39 @@ TransformedScalarCoefficient::TransformedScalarCoefficient(std::shared_ptr<mfem:
 {
 }
 
-double TransformedScalarCoefficient::Eval(mfem::ElementTransformation& T, const mfem::IntegrationPoint& ip)
+TransformedScalarCoefficient::TransformedScalarCoefficient(std::unique_ptr<mfem::Coefficient>&& u1,
+                                                           std::function<double(const double)>  func)
+    : mfem::Coefficient(), u1_(std::move(u1)), mono_function_(func), bi_function_(nullptr)
 {
-  double temp = s1_->Eval(T, ip);
+}
 
-  if (mono_function_) {
-    return mono_function_(temp);
+TransformedScalarCoefficient::TransformedScalarCoefficient(std::unique_ptr<mfem::Coefficient>&&              u1,
+                                                           std::unique_ptr<mfem::Coefficient>&&              u2,
+                                                           std::function<double(const double, const double)> func)
+    : mfem::Coefficient(), u1_(std::move(u1)), u2_(std::move(u2)), mono_function_(nullptr), bi_function_(func)
+{
+}
+
+double TransformedScalarCoefficient::Eval(mfem::ElementTransformation& T, const mfem::IntegrationPoint& ip)
+{
+  if (s1_ != nullptr) {
+    double temp = s1_->Eval(T, ip);
+
+    if (mono_function_) {
+      return mono_function_(temp);
+    } else {
+      double temp2 = s2_->Eval(T, ip);
+      return bi_function_(temp, temp2);
+    }
   } else {
-    double temp2 = s2_->Eval(T, ip);
-    return bi_function_(temp, temp2);
+    double temp = u1_->Eval(T, ip);
+
+    if (mono_function_) {
+      return mono_function_(temp);
+    } else {
+      double temp2 = u2_->Eval(T, ip);
+      return bi_function_(temp, temp2);
+    }
   }
 }
```

Comment:
The coefficient class seems like such a clunky way to shoehorn simple calculations into the integrators, I look forward to Julian's work making this sort of thing obsolete!

Reply:
me too!!!